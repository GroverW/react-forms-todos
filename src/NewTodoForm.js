import React, { useState } from 'react';

const INITIAL_STATE = ""

const NewTodoForm = ({ addTodo }) => {
  const [formData, setFormData] = useState(INITIAL_STATE);
  const handleSubmit = (evt) => {
    evt.preventDefault();
    addTodo(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = (evt) => {
    setFormData(evt.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">New Task:</label>
      <input id="task" name="newTask" type="text" onChange={handleChange} value={formData}/>
      <button>Add!</button>
    </form>
  );
};

export default NewTodoForm;