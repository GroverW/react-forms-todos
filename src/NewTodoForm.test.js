import React from 'react';
import { render, fireEvent, queryByDisplayValue } from "@testing-library/react";
import NewTodoForm from './NewTodoForm';


it('renders without crashing', () => {
  render(<NewTodoForm />);
});

it('matches the snapshot', () => {
  const {asFragment} = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it('updates user inputs in form value', () => {
  const {queryByText, queryByLabelText, queryByDisplayValue } = render(<NewTodoForm addTodo={() => {}}/>);

  let task = queryByLabelText("New Task:");
  let addBtn = queryByText("Add!");

  fireEvent.change(task, {target: {value: "Task 1"}});

  expect(queryByDisplayValue("Task 1")).toBeInTheDocument();

  fireEvent.click(addBtn);
  expect(queryByDisplayValue("Task 1")).not.toBeInTheDocument();
});