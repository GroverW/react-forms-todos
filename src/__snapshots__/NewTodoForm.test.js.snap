// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      New Task:
    </label>
    <input
      id="task"
      name="newTask"
      type="text"
      value=""
    />
    <button>
      Add!
    </button>
  </form>
</DocumentFragment>
`;
